mininet/node.py

```diff
@@ -783,14 +783,37 @@ def __init__(
 
         # let's initially set our resource limits
         self.update_resources(**self.resources)
-        # self.updateCpuLimit(cpu_quota=self.resources.get('cpu_quota'),
-        #                     cpu_period=self.resources.get('cpu_period'),
-        #                     cpu_shares=self.resources.get('cpu_shares'),
-        #                     )
-        # self.updateMemoryLimit(mem_limit=self.resources.get('mem_limit'),
-        #                        memswap_limit=self.resources.get('memswap_limit')
-        #                        )
 
+        # Containernet ignores the CMD field of the Dockerfile.
+        # Lets try to load it here and manually execute it once the
+        # container is started and configured by Containernet:
+        if not kwargs.get("no_cmd_field_execution", False):
+            cmd_field = self.get_cmd_field(self.dimage)
+            if cmd_field is not None:
+                cmd_field.append("&")  # put to background (works, but not nice)
+                info("{}: running CMD: {}\n".format(name, cmd_field))
+                self.cmd(" ".join(cmd_field))
```

Comment:
this probably should quote the individual cmd parts (otherwise spaces in `cmd_field` breaks an argument into two... However, I am not sure what the right way to quote them is since the quoting character might collide with characters within the field itself... :confused:

Reply:
Good point. I am also not sure how the list of CMDs is meant by Docker. Maybe one should even execute each `cmd_field` individually ... one after each other. Do we have more examples for those CMD lists of commonly used containers available?

Reply:
In general the CMD field is applied as parameters to the ENTRYPOINT, but never as individual commands.

Some examples:
* `ENTRYPOINT` not set
* `CMD=/bin/bash`
-> `bash` is executed

* `ENTRYPOINT=["/bin/bash", "-c"]`
* `CMD=["echo", "test"]`
-> `/bin/bash -c echo test` is executed

Whether you specify CMD/ENTRYPOINT as `["foo", "bar"]` or as `foo bar` should not matter (but I am not sure how the inspect call returns the individual cases).

Reply:
Ok did some tests: In general, only the last `CMD` definition of a Dockerfile is passed on. So there is no way to define multiple `CMD` fields in a Dockerfile. 

Also Docker automatically escapes quotes:

Dockerfile `CMD ls -I "*.py" -l | grep 'lib'`
Gives: `[u'/bin/sh', u'-c', u'ls -I "*.py" -l | grep \'lib\'', '&']`

This is then correctly executed. So I would assume the current version should work for most cases. Should be ok for an experimental feature. We would need to see if someone finds something which break with this approach, no?

Another thing is the support for `ENTRYPOINT` which could also be added like this.

Reply:
Just did a quick test with a Dockerfile that contains an `ENTRYPOINT` statement. This still blocks Containernet completely since it is not overwritten by default. Thus it will be a biger change. I will move this to a separated issue.

Reply:
#103